client.go

```diff
@@ -364,6 +364,48 @@ func (h hookLogger) Printf(s string, args ...interface{}) {
 	h.Info(fmt.Sprintf(s, args...))
 }
 
+// ContextLogger is an interface that provides methods for logging
+// with context. The methods accept a context.Context, a message
+// string and a variadic number of key-value pairs.
+type ContextLogger interface {
+	ErrorContext(ctx context.Context, msg string, keysAndValues ...interface{})
+	InfoContext(ctx context.Context, msg string, keysAndValues ...interface{})
+	DebugContext(ctx context.Context, msg string, keysAndValues ...interface{})
+	WarnContext(ctx context.Context, msg string, keysAndValues ...interface{})
+}
+
+// contextLogger adapts an ContextLogger to Logger for use by the existing hook functions
+// without changing the API.
+type contextLogger struct {
+	logger ContextLogger
+}
+
+// NewContextLogger returns a new contextLogger
+// for example: NewContextLogger(slog.Default())
+func NewContextLogger(logger ContextLogger) ContextLogger {
```

Comment:
A constructor is added here to make it easier for users to use. Due to the existing methods in LeveledLogger, ordinary loggers such as slog cannot be directly passed in and will be identified as LeveledLogger. The constructor is used here for packaging, which makes it easier for external users to use ContextLogger.

Reply:
I think `NewContextLogger` now needs `ctx` parameter.

Reply:
I remove it, now you only need to pass in `slog` directly to use it. You don't need to use a constructor to wrap it. If you use another logger externally, you can flexibly wrap it.
What do you think?

Reply:
Ok

Reply:
I think this is fine, yes.